superchain/internal/codegen/main.go

```diff
@@ -0,0 +1,120 @@
+	Name            string   `json:"name" toml:"name"`
+	Identifier      string   `json:"identifier" toml:"identifier"`
+	ChainId         uint64   `json:"chainId" toml:"chainId"`
+	RPC             []string `json:"rpc" toml:"rpc"`
+	Explorer        []string `json:"explorers" toml:"explorers"`
+	SuperchainLevel uint     `json:"superchain_level" toml:"superchain_level"`
+	Parent          Parent   `json:"parent" toml:"parent"`
+}
+
+type Parent struct {
+	Type    string   `json:"type" toml:"type"`
+	Chain   string   `json:"chain" toml:"chain"`
+	Bridges []string `json:"bridge,omitempty" toml:"bridges,omitempty"`
+}
+
+func main() {
+	allChains := make([]ChainEntry, 0)
+	for sc := range Superchains {
+		standardChains := make([]ChainEntry, 0)
+		frontierChains := make([]ChainEntry, 0)
+		for _, chainId := range Superchains[sc].ChainIDs {
+			chain := OPChains[uint64(chainId)]
+			if chain == nil {
+				log.Fatalf("cannot find chain with id %d", chainId)
+			}
+			chainEntry := ChainEntry{
+				Name:            chain.Name,
+				Identifier:      chain.Identifier(),
+				ChainId:         chain.ChainID,
+				RPC:             []string{chain.PublicRPC},
+				Explorer:        []string{chain.Explorer},
+				SuperchainLevel: uint(chain.SuperchainLevel),
+				Parent:          Parent{"L2", chain.Superchain, []string{}},
+			}
+			switch chain.SuperchainLevel {
+			case Standard:
+				standardChains = append(standardChains, chainEntry)
+			case Frontier:
+				frontierChains = append(frontierChains, chainEntry)
+			default:
+				panic(fmt.Sprintf("unknown SuperchanLevel %d", chain.SuperchainLevel))
+			}
+		}
+		allChains = append(allChains, standardChains...)
+		allChains = append(allChains, frontierChains...)
+	}
+
+	fmt.Printf("Found %d chains...\n", len(allChains))
+
+	currentDir := currentDir()
+
+	allChainsBytes, err := json.MarshalIndent(allChains, "", "  ")
+	if err != nil {
+		panic(err)
+	}
+	err = os.WriteFile(filepath.Join(currentDir, "../../configs/chainList.json"), allChainsBytes, 0o644)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Wrote chainList.json file")
+
+	var buf bytes.Buffer
+	allChainsForTOML := map[string]([]ChainEntry){"chains": allChains}
+	if err := toml.NewEncoder(&buf).Encode(allChainsForTOML); err != nil {
+		fmt.Println("Error encoding TOML:", err)
+		return
+	}
+	err = os.WriteFile(filepath.Join(currentDir, "../../configs/chainList.toml"), buf.Bytes(), 0o644)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Wrote chainList.toml file")
+
+	addresses()
+}
+
+func currentDir() string {
+	_, currentFilePath, _, ok := runtime.Caller(0)
+	if !ok {
+		fmt.Println("Error: Unable to determine the current file path")
+		os.Exit(1)
+	}
+	return filepath.Dir(currentFilePath)
+}
+
+func addresses() {
+	addressList := make(map[string]AddressList)
+	for id := range OPChains {
+		list := Addresses[id]
+		addressList[strconv.Itoa(int(id))] = *list
+	}
+	addressListBytes, err := json.MarshalIndent(addressList, "", " ")
```

Comment:
This new address writer sets all non-existent contracts to the zero address. Non-existent contracts seem to have been omitted before, which I think is better?

Reply:
I spent some time on this in the end. The zero addresses being written does stand out somewhat. It is actually out of scope for this PR, so I have reinstated the existing behaviour to unblock us.

I think it is worth coming back to, though. Thoughts and solutions I discarded for now:

1. Writing out zeros for "missing" addresses reflects the behaviour of the Go bindings. I wonder if we should use a pointer to an address inside `AddressList`, especially for contracts which we don't expect to always be there (i.e. optional fields). It seems to better reflect the underlying data -- especially if we ever got into a situation where a zero address was actually a meaningful value for us, it is a meaningful value in the EVM. I actually tried this type change out but it quickly snowballs and is a breaking change for downstream packages. 

2. The `omitempty` directive only works for "empty" fields which are being marshalled, and `[20]byte{}` is not considered empty. It is possible to get around this by writing custom marshalling code, but it is pretty nasty and lots of boilerplate.

Reply:
ok bummer that it created so much complexity...